include/rocksdb/options.h

```diff
@@ -847,6 +847,18 @@ struct DBOptions {
   //
   // Dynamically changeable through SetDBOptions() API.
   bool avoid_flush_during_shutdown = false;
+
+  // If enabled it uses two queues for writes, one for the ones with
+  // disable_memtable and one for the ones that also write to memtable. This
+  // allows the memtable writes not to lag behind other writes. It can be used
+  // to optimize MySQL 2PC in which only the commits, which are serial, write to
+  // memtable.
+  bool separate_queue_for_memtable_writes = false;
+
+  // If true WAL is not flushed automatically after each write. Instead it
+  // relies on manual invocation of FlushWAL to write the WAL buffer to its
+  // file.
+  bool manual_wal_flush = false;
```

Comment:
1. In theory the other queue should help: without queue 128 prepare threads competing with commit on wal_mutex_ but with queue only 1 thread does that. In practice I have seen mixed results: in some it improved throughput by 3-4k.
2. Without queue other part of the code that wanted to be the sole writer would still need to hold a mutex. Combination of mutex and enter queue unbatched was a bit ugly. Instead now they enter two queues.